kmelia/kmelia-war/src/main/java/org/silverpeas/components/kmelia/ui/ResultSearchRenderer.java

```diff
@@ -133,7 +134,7 @@ private void setSpecificAttributes(SearchResultContentVO searchResult,
       template.setAttribute("pubContent", pubDetail.getContentPagePath());
     }
     String componentId = silverResult.getInstanceId();
-    String id = silverResult.getId();
+    String id = silverResult.isLinked() ? silverResult.getLinkedResourceId() : silverResult.getId();
```

Comment:
There are a lot of ternary expressions like this one in CORE and COMPONENTS.
Maybe the bean model should provide a signature to centralize it? `getPrincipalId()` or `getMainId()` for example?

Reply:
Yes, I thought of that to avoid such a duplication of code but in fact, this ternary expression is a rule of components to create the correct link to the contribution. Whereas this rule is common, currently, to the components, it isn't the responsibility of the search result bean.

Reply:
OKi